runtime/caml/weak.h

```diff
@@ -18,130 +18,30 @@
 #ifndef CAML_WEAK_H
 #define CAML_WEAK_H
 
+#ifdef CAML_INTERNALS
+
```

Comment:
Is the additional of this guard correct (I expect so, just checking!)

Reply:
This guard is also present on trunk (lower down the file) after the ephemeron C API. Multicore doesn't yet support the ephemeron C API. But it turns out that the guard is wrongly nested wrt `ifdef __cplusplus. I'll fix this shortly.